opentelemetry/daemonset/clusterrole.yaml

```diff
@@ -0,0 +1,8 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otel-collector
+rules:
+- apiGroups: [""]
+  resources: ["pods", "namespaces"]
```

Comment:
a note here why these perms are needed would be useful to know.

Reply:
There is a note in the `README.md`:
> The `ClusterRole` and `ClusterRoleBinding` are [needed by the `k8sattributesprocessor`](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/processor/k8sattributesprocessor#role-based-access-control).